### ISSUE TITLE
about wgan's loss function

### ISSUE THREAD

**Issue by dingling00** [shekkizh/WassersteinGAN.tensorflow#7]
according to the wgan paper , when training the critic, we need to maximize Er(f(x)) - Eg(f(x)), and when training the generator, we need to minimize Er(f(x)) - Eg(f(x)), that is to say minimize - Eg(f(x)), but in the code, wgan's loss use  self.discriminator_loss = tf.reduce_mean(logits_real - logits_fake) , and self.gen_loss = tf.reduce_mean(logits_fake) , so I'm confused.

**Comment by dingling00** [comment]
what I'm confused is whether use negative, because according to tensorflow's API , we use optimizer.compute_gradients(loss), the loss is what we what to minimize, but the loss in WGAN code is to maximize, so I'm confused

**Comment by yifita** [comment]
I have the same question here.
I think the one should negate the discriminator loss

**Comment by dingling00** [comment]
I ask the question on Reddit, the website is: https://www.reddit.com/r/MachineLearning/comments/5qxoaz/r_170107875_wasserstein_gan/
and ajmooch answer me, this is what he said:  'I suspect this is just a case of sign flippage--if you just switch conventions so that your goal is to make the critic output highly negative values for real samples and highly positive values for fake samples (and change the generator's objective accordingly) then it still works the same; since there's no nonlinearity we can flip those signs with impunity'.
I think maybe he is right, if we change the sign of the discriminator loss, we are just minmax W(pr,pg) or minmax W(pg, pr), so if we change the generator's objective accordingly, we are doing the same thing.

**Comment by yifita** [comment]
ah you are right. I didn't notice the g_loss is reduce_mean(fake) instead of -reduce_mean(fake)

**Comment by shekkizh** [comment]
Duplicate of #6

**Comment by shamanez** [comment]
Well simply understanding is according to WGAN loss D should try to maximize the distance between Real and Fake Data. This will be completed by the logits given by the discriminator neural net. So somehow we need to make this tractable. So what the Generator does is try to reduce the minus value of logits generated for gen output from the discriminator. So basically Gen will try to generate more and more plus values. But discriminator is trying to make logits for gen output more minus and maximize the plus value of discriminator logit output for real dat

**Comment by nicholasly** [comment]
I have a problem: the loss of the generator is descending from positive to negative but I have negated the generator loss. Is the negative value of the generator loss bad for the optimization?

**Comment by 478731807** [comment]
@nicholasly I also met with the same issue. The WGAN was trained for timeseries data. The D and G loss all become negative after several epoch. What should I do? Have you figure out why? Thx

**Comment by nicholasly** [comment]
I currently think a negative value is acceptable when considering the weight changes in the network during the backpropagation. But I have not figured out how to monitor the training if the loss can converge to the value samller than zero.

**Comment by 478731807** [comment]
@nicholasly But why this could happen? Maybe I use the timeseries data with different length and feature numbers? Or the 2D CNN structure is not working under my data circumstance? lol...I am tring to figure out

**Comment by LexieYang** [comment]
@nicholasly Have you figure out why the loss of the generator is descending from positive to negative? Is that normal?

**Comment by Kaede93** [comment]
@LexieYang @nicholasly 
Have you guys figure out why the Wasserstein loss can be negative? Even thought the definition of the Wasserstein loss makes it can be negative, but what's the negative value stand for?
